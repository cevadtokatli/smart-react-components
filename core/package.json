{
  "name": "@smart-react-components/core",
  "version": "2.0.0-beta-0",
  "license": "MIT",
  "main": "index.js",
  "description": "SRC core includes the theme, Styled properties, and Element components.",
  "keywords": [
    "javascript",
    "typescript",
    "react",
    "styled-components"
  ],
  "author": {
    "name": "Cevad Tokatlı",
    "email": "cevadtokatli@hotmail.com"
  },
  "contributors": [
    {
      "name": "Cemil Tokatlı",
      "email": "cemil@tokatli.me"
    }
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/cevadtokatli/smart-react-components/tree/v2/core"
  },
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "yarn clear && tsc --outDir lib && yarn copy",
    "clear": "rm -rf lib",
    "copy": "cp package.json lib/package.json && cp README.md lib/README.md && cp ../LICENSE lib/LICENSE",
    "dev": "tsc -w --outDir lib",
    "lint": "eslint src",
    "release": "../scripts/release.sh core",
    "test": "jest"
  },
  "devDependencies": {
    "@types/color": "3.0.3",
    "@types/enzyme": "3.10.12",
    "@types/jest": "29.0.1",
    "@types/react": "18.0.19",
    "@types/react-dom": "18.0.6",
    "@types/styled-components": "5.1.26",
    "@wojtekmaj/enzyme-adapter-react-17": "0.6.7",
    "enzyme": "3.11.0",
    "eslint": "8.0.1",
    "jest": "29.0.3",
    "jsdom": "20.0.0",
    "jsdom-global": "3.0.2",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "styled-components": "5.3.5",
    "ts-jest": "29.0.0",
    "typescript": "4.8.3"
  },
  "dependencies": {
    "color": "4.2.3"
  },
  "peerDependencies": {
    "react": "^18.2.0",
    "styled-components": "^5.3.5"
  }
}
