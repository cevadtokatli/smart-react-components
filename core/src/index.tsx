export * from './theme'
export { default as createTheme } from './theme'
export * from './styled-props'
export * from './Element'
