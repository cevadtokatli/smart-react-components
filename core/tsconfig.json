{
  "extends": "../tsconfig.json",
  "include": ["src/**/*.tsx"],
  "exclude": ["node_modules", "src/**/*.test.tsx"]
}
