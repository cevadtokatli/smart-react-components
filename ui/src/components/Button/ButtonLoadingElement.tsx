import Div from '@smart-react-components/core/Element/Div'
import { StyledProps } from '@smart-react-components/core/styled-props'
import styled from 'styled-components'

interface Props extends StyledProps {
  hasAnimation: boolean
}

export default styled(Div)<Props>(({ theme, hasAnimation }: Props) => `
  align-items: center;
  display: flex;
  height: 100%;
  justify-content: center;
  left: 0;
  position: absolute;
  width: 100%;
  z-index: ${theme.$.zIndex.buttonLoading};

  ${hasAnimation
    ? `
      &.src-button-loading-show {
        opacity: 0;
        transform: scale(0);
      }

      &.src-button-loading-show-active {
        transition: 300ms 0s ease-in-out;
        transition-property: opacity, transform;
        opacity: 1;
        transform: scale(1);
      }

      &.src-button-loading-hide {
        opacity: 1;
        transform: scale(1);
      }

      &.src-button-loading-hide-active {
        transition: 300ms 0s ease-in-out;
        transition-property: opacity, transform;
        opacity: 0;
        transform: scale(0);
      }
    `
    : ''
  }
`)
