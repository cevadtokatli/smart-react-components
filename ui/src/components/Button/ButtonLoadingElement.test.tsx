import { wrapTheme } from '@smart-react-components/core/test'
import { render } from '@testing-library/react'
import React from 'react'
import ButtonLoadingElement from './ButtonLoadingElement'

describe('<ButtonLoadingElement />', () => {
  it('should render component', () => {
    const { asFragment } = render(wrapTheme(<ButtonLoadingElement hasAnimation={false} />))
    expect(asFragment()).toMatchSnapshot()
  })

  it('should have animation', () => {
    const { asFragment } = render(wrapTheme(<ButtonLoadingElement hasAnimation />))
    expect(asFragment()).toMatchSnapshot()
  })
})
