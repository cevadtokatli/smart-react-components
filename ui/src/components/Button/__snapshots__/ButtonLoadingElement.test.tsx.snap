// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonLoadingElement /> should have animation 1`] = `
<DocumentFragment>
  .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  left: 0;
  position: absolute;
  width: 100%;
  z-index: 10;
}

.c0.src-button-loading-show {
  opacity: 0;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
}

.c0.src-button-loading-show-active {
  -webkit-transition: 300ms 0s ease-in-out;
  transition: 300ms 0s ease-in-out;
  -webkit-transition-property: opacity,-webkit-transform;
  -webkit-transition-property: opacity,transform;
  transition-property: opacity,transform;
  opacity: 1;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c0.src-button-loading-hide {
  opacity: 1;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c0.src-button-loading-hide-active {
  -webkit-transition: 300ms 0s ease-in-out;
  transition: 300ms 0s ease-in-out;
  -webkit-transition-property: opacity,-webkit-transform;
  -webkit-transition-property: opacity,transform;
  transition-property: opacity,transform;
  opacity: 0;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
}

<div
    class=" c0"
  />
</DocumentFragment>
`;

exports[`<ButtonLoadingElement /> should render component 1`] = `
<DocumentFragment>
  .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  left: 0;
  position: absolute;
  width: 100%;
  z-index: 10;
}

<div
    class=" c0"
  />
</DocumentFragment>
`;
